### PR TITLE
bank: added user logout

### LINKED ISSUE #43: bank: user logout
Users should be able to logout and terminate their sessions by will for obvious security reasons.

### PR DESCRIPTION
**Pull request by gadjevb**
Added user logout, so that users have the ability to terminate their sessions at will, for obvious security reasons.

Fixes #43

### CODE REVIEW

File: src/main/java/com/clouway/bank/adapter/http/LogoutService.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.google.sitebricks.At;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.headless.Service;
+import com.google.sitebricks.http.Get;
+
+/**
+ * @author Borislav Gadjev <gadjevb@gmail.com>
+ */
+@Service
+@At("/v1/transactions")
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/LogoutService.java]: ??

File: src/main/java/com/clouway/bank/adapter/http/LogoutService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private UserSecurity security;
+
+    @Inject
+    public LogoutService(SessionRepository sessionRepository, UserSecurity security) {
+        this.sessionRepository = sessionRepository;
+        this.security = security;
+    }
+
+    @Get
+    public Reply<?> logUserOut() {
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/LogoutService.java]: logout

File: src/main/java/com/clouway/bank/adapter/http/LogoutService.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        User currentUser = security.currentUser().get();
```
Comment (mgenov) [inline src/main/java/com/clouway/bank/adapter/http/LogoutService.java]: What happens if session was expired?

**Comment by gadjevb** [comment]
@mgenov made the according changes!

File: src/test/java/com/clouway/bank/adapter/http/LogoutServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  private UserSecurity security = context.mock(UserSecurity.class);
+  private LogoutService logoutService = new LogoutService(sessionRepository, security);
+
+  @Test
+  public void userTerminatesSession() {
+    context.checking(new Expectations() {{
+      oneOf(security).currentUser();
+      will(returnValue(Optional.of(new User("123", "Borislav", "123456"))));
+
+      oneOf(sessionRepository).terminateSession(with(any(User.class)));
```
Comment (mgenov) [inline src/test/java/com/clouway/bank/adapter/http/LogoutServiceTest.java]: terminateUserSession

**Comment by gadjevb** [comment]
@mgenov fixed the name of the method!

**Pull request merged**